IPython/terminal/shortcuts.py

```diff
@@ -123,6 +124,18 @@ def _following_text():
         _following_text_cache[pattern] = condition
         return condition
 
+    @Condition
+    def not_inside_unclosed_string():
+        app = get_app()
+        ver = f"{sys.version_info.major}.{sys.version_info.minor}"
+        g = parso.load_grammar(version=ver)
+        parser = g.parse(app.current_buffer.document.current_line_before_cursor)
+        for e in g.iter_errors(parser):
+            # check for error scanning string literal
+            if e.code == 901:
```

Comment:
Are the error code stable ?

Reply:
that's been a [SyntaxRule](https://github.com/davidhalter/parso/blame/86f3f1096b1126df8b8c914a54a4951a30e67eaa/parso/python/errors.py#L505) for five years, so probably.

So could be: 
```
if e.code == SyntaxRule.code:
````

Reply:
They are stable, they’re the ones defined in PEP8, but actually it seems wiser to check the error message instead of the code because 901 matches other syntax errors too.